pkg/operator/ceph/cluster/cephstatus_test.go

```diff
@@ -197,3 +203,104 @@ func Test_cephStatusChecker_conditionMessageReason(t *testing.T) {
+	}
+
+	c := newCephStatusChecker(context, clusterInfo, &cephv1.ClusterSpec{})
+
+	pod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "stuck-pod",
+			Namespace: clusterName.Namespace,
+			Labels: map[string]string{
+				"rook_cluster": clusterName.Name,
+			},
+		},
+	}
+	pod.Spec.NodeName = "node0"
+	_, err := context.Clientset.CoreV1().Pods(clusterName.Namespace).Create(ctx, &pod, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	// Set the node to NotReady state
+	nodes, err := context.Clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	assert.NoError(t, err)
+	for _, node := range nodes.Items {
+		node.Status.Conditions[0].Status = v1.ConditionFalse
+		localnode := node
+		_, err := context.Clientset.CoreV1().Nodes().Update(ctx, &localnode, metav1.UpdateOptions{})
+		assert.NoError(t, err)
+	}
+
+	// There should be no error
+	err = c.forceDeleteStuckRookPodsOnNotReadyNodes()
+	assert.NoError(t, err)
+
+	// The pod should still exist since its not deleted.
+	p, err := context.Clientset.CoreV1().Pods(clusterInfo.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.NotNil(t, p)
+
+	// Add a deletion timestamp to the pod
+	pod.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	_, err = clientset.CoreV1().Pods(clusterName.Namespace).Update(ctx, &pod, metav1.UpdateOptions{})
+	assert.NoError(t, err)
+
+	// There should be no error as the pod is deleted
+	err = c.forceDeleteStuckRookPodsOnNotReadyNodes()
+	assert.NoError(t, err)
+
+	// The pod should be deleted since the pod is marked as deleted and the node is in NotReady state
+	_, err = clientset.CoreV1().Pods(clusterName.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	assert.Error(t, err)
+	assert.True(t, kerrors.IsNotFound(err))
+}
+
+func TestGetRookPodsOnNode(t *testing.T) {
+	ctx := context.TODO()
+	clientset := optest.New(t, 1)
+	clusterInfo := client.NewClusterInfo("test", "test")
+	clusterName := clusterInfo.NamespacedName()
+	context := &clusterd.Context{
+		Clientset: clientset,
+	}
+
+	c := newCephStatusChecker(context, clusterInfo, &cephv1.ClusterSpec{})
+	labels := []struct {
+		key   string
+		value string
+	}{
+		{"rook_cluster", clusterName.Name},
+		{"app", "csi-rbdplugin-provisioner"},
+		{"app", "csi-rbdplugin"},
+		{"app", "csi-cephfsplugin-provisioner"},
+		{"app", "csi-cephfsplugin"},
+		{"app", "rook-ceph-operator"},
+	}
+
+	for i, label := range labels {
+		pod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("pod-%d", i),
+				Namespace: clusterName.Namespace,
+			},
+		}
+		pod.ObjectMeta.Labels = map[string]string{label.key: label.value}
+		pod.Spec.NodeName = "node0"
+		_, err := context.Clientset.CoreV1().Pods(clusterName.Namespace).Create(ctx, &pod, metav1.CreateOptions{})
+		assert.NoError(t, err)
+
+	}
+
+	// The same pod sholud appear thrice as it has three labels
+	pods, err := c.getRookPodsOnNode("node0")
+	assert.NoError(t, err)
+	assert.Equal(t, 6, len(pods))
```

Comment:
Don't just check the length, but also check that the correct pods are returned.

Reply:
Using labels to check for the correct pods.

Reply:
I think it would make a better test to verify that the correct pods are in the list by-name and not by label. This will also verify that pods don't appear in the list multiple times. The test already knows what names the pods should have (`pod-0` thru `pod-6`). Otherwise, the test is also testing itself with circular logic.